### PR TITLE
#320  Tag tests

### LINKED ISSUE #320: category.es6:8-10: Take selenium tests for category page....
The puzzle `187-bccbbf6e` from #187 has to be resolved:

https://github.com/fidals/stroyprombeton/blob/4f1b5b3c8da15d2450bfa77e9c229b713213c546/front/js/category.es6#L8-L10

The puzzle was created by duker33 on 24-Sep-18. 

Estimate: 120 minutes,     

If you have any technical questions, don't ask me, submit new tickets instead. The task will be "done" when the problem is fixed and the text of the puzzle is _removed_ from the source code. Here is more about [PDD](http://www.yegor256.com/2009/03/04/pdd.html) and [about me](http://www.yegor256.com/2017/04/05/pdd-in-action.html).

### PR DESCRIPTION
**Pull request by duker33**
Closes #320

### CODE REVIEW

File: stroyprombeton/management/commands/test_db.py [hunk 1]
```diff
@@ -214,9 +214,13 @@ def create_products(count, categories, tags_):
                     for tag in tags_:
                         product.tags.add(tag)
 
+        # group together tags from different tag_groups
```
Comment (artemiy312) [inline stroyprombeton/management/commands/test_db.py]: The comment below will be enough. Decide

File: stroyprombeton/management/commands/test_db.py [hunk 2]
```diff
@@ -214,9 +214,13 @@ def create_products(count, categories, tags_):
...
+        # [('1 м', '2 кг', '72 %'), ('2 м', '3 кг', '146 %')]
         zipped_tags = list(zip(*tags))
         # Create 25 products for
         # tests_selenium.models.Category.Page.test_load_more_hidden_in_fully_loaded_categories
-        create_products(count=25, categories=parents[4:], tags_=zipped_tags[0])
+        create_products(count=25, categories=parents[5:], tags_=zipped_tags[0])
+        # some products should not contain any tag
+        create_products(count=25, categories=parents[4:4], tags_=[])
```
Comment (artemiy312) [inline stroyprombeton/management/commands/test_db.py]:
```suggestion
        create_products(count=25, categories=parents[4:5], tags_=[])
```

`[4:4]` produces an empty list

File: stroyprombeton/tests/tests_selenium.py [hunk 3]
```diff
@@ -1,18 +1,43 @@
...
 from selenium.webdriver.common.action_chains import ActionChains
 from selenium.webdriver.common.keys import Keys
 from selenium.webdriver.common.by import By
-from selenium.webdriver.support import expected_conditions as EC
+from selenium.webdriver.support import expected_conditions as EC, ui
 
 from pages.models import Page
 
-from stroyprombeton.models import Category, Product
+# @todo #320:15m - Do the whole `models` import in tests_selenium
```
Comment (artemiy312) [inline stroyprombeton/tests/tests_selenium.py]: Import the `stroyprombeton.models` package as `stb_models` for the whole project.

File: stroyprombeton/tests/tests_selenium.py [hunk 4]
```diff
@@ -1,18 +1,43 @@
...
 
 
+def reverse_catalog_url(
+    url: str,
+    route_kwargs: dict,
+    tags: TagQuerySet=None,
+    sorting: int=None,
+    query_string: dict=None,
+) -> str:
+    from urllib.parse import urlencode
```
Comment (artemiy312) [inline stroyprombeton/tests/tests_selenium.py]: Move an import to the top

File: stroyprombeton/tests/tests_views.py [hunk 5]
```diff
@@ -195,6 +207,25 @@ def test_inactive_product_not_in_category(self):
         response = self.client.get(reverse('category', args=(test_product.category_id,)))
         self.assertNotIn(test_product, response.context['products'])
 
+    # @todo #320:60m Take from SE CatalogTags tests
```
Comment (artemiy312) [inline stroyprombeton/tests/tests_views.py]: What should we take?

File: stroyprombeton/tests/tests_selenium.py (hunk 3 above)
Comment (duker33) [reply stroyprombeton/tests/tests_selenium.py]: @artemiy312 , it's too long. I suggest this:
`Import `models` to it's own var`

**Pull request merged**